### PR TITLE
✨ Add `allowed` option for inline/block attributes

### PR DESCRIPTION
**Pull request by joker21663**
adding allowed attributes to protect against XSS attacks.
An example of such an attack:
asd{onerror="alert(1)"}

### CODE REVIEW

File: mdit_py_plugins/attrs/index.py
Comment (chrisjsewell) [inline mdit_py_plugins/attrs/index.py:25]: just condense these

```suggestion
    allowed_attributes: Sequence[str] | None = None,
```

Comment (joker21663) [reply mdit_py_plugins/attrs/index.py]: brought it together

File: mdit_py_plugins/attrs/index.py
Comment (chrisjsewell) [inline mdit_py_plugins/attrs/index.py:71]: err, I'm a little hesitant to add in intentional exceptions:
Markdown is generally designed to be fault-tolerant, i.e. parsing never fails. This is e.g. the case for HTML tags; unsecure ones are just ignored, rather than excepted.

I would simply remove the unsecure attributes from the `attrs`, then store them on e.g. `attr_token.meta["insecure"_attrs"]`.
Then external programs could access them, if they desire to warn about them,
but otherwise parsing will succeed

**Review by chrisjsewell — CHANGES_REQUESTED** [review]

File: mdit_py_plugins/attrs/index.py
Comment (joker21663) [reply mdit_py_plugins/attrs/index.py]: fixed

File: mdit_py_plugins/attrs/index.py
Comment (joker21663) [reply mdit_py_plugins/attrs/index.py]: fixed

File: mdit_py_plugins/attrs/index.py
Comment (chrisjsewell) [inline mdit_py_plugins/attrs/index.py:68]:
```suggestion
            if allowed_attributes is not None and (disallowed := {
                    k: v for k, v in attrs.items() if k not in allowed_attributes
                }):
                token.meta["insecure_attrs"] = dissalowed
                attrs = {k: v for k, v in attrs.items() if k in allowed_attributes}
```

you were tryting to take the insecure after you'd already removed them 😅

File: mdit_py_plugins/attrs/index.py
Comment (chrisjsewell) [inline mdit_py_plugins/attrs/index.py:68]:
```suggestion
            if allowed_attributes is not None and (disallowed := {
                    k: v for k, v in attrs.items() if k not in allowed_attributes
                }):
                token.meta["insecure_attrs"] = disallowed
                attrs = {k: v for k, v in attrs.items() if k in allowed_attributes}
```

Comment (joker21663) [reply mdit_py_plugins/attrs/index.py]: Are these all mistakes? or did I screw up somewhere else?

**Comment by chrisjsewell** [comment]
@joker21663 I reworked your PR 😅; to also work for blocks, and add a test

If you wanna have a look and comment before I merge

**Review by chrisjsewell — APPROVED** [review]

**Comment by joker21663** [comment]
> @joker21663 I reworked your PR 😅; to also work for blocks, and add a test
> 
> If you wanna have a look and comment before I merge

I'm not a programmer. I am a user who was forbidden by the security service to use your plugin without modification). I believe your code and I'm waiting for the merge and the new version so that I can use the plugin and make life easier for our accounting department. Thank you.

**Pull request merged**
